### PR TITLE
Implement XL layer

### PR DESCRIPTION
**Pull request by harshavardhana**

### CODE REVIEW

File: disk.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+	}
+
+	info, e := disk.GetInfo(diskPath)
+	if e != nil {
+		return nil, e
+	}
+	disk := storageDisk{
+		diskPath:    diskPath,
+		fsInfo:      info,
+		minFreeDisk: 5, // Minimum 10% disk should be free.
```
Comment (balamurugana) [inline disk.go]: As we set `5` here, the comment says 10%
it looks confusing

File: disk.go [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+	}
+	return disk, nil
+}
+
+var errDiskPathFull = errors.New("Disk path full.")
+var errVolumeExists = errors.New("Volume already exists.")
+
+// Make a volume entry.
+func (s storageDisk) MakeVol(volume string) error {
```
Comment (balamurugana) [inline disk.go]: What is valid volume name?

**Comment by harshavardhana** [comment]
> As we set 5 here, the comment says 10%
> it looks confusing

Yeah this is old, will clean it. 

> What is valid volume name?

All of it is valid.

File: disk.go [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+	di, e := disk.GetInfo(s.diskPath)
+	if e != nil {
+		return e
+	}
+
+	// Remove 5% from total space for cumulative disk space used for journalling, inodes etc.
+	availableDiskSpace := (float64(di.Free) / (float64(di.Total) - (0.05 * float64(di.Total)))) * 100
```
Comment (balamurugana) [inline disk.go]: we need to use storageDisk.minFreeDisk here

**Pull request closed without merging**